src/coredump/coredump.c

```diff
@@ -1659,11 +1659,11 @@ static int gather_pid_mount_tree_fd(const Context *context) {
 
         /* Don't bother preparing environment if we can't pass it to libdwfl. */
 #if !HAVE_DWFL_SET_SYSROOT
-        return -EBADF;
+        return -EOPNOTSUPP;
 #endif
 
         if (!arg_access_container)
-                return -EBADF;
+                return -EOPNOTSUPP;
```

Comment:
but this doesn't. user configuratoin should not result in EOPNOTSUPP. Use something such as EHOSTDOWN maybe?

Reply:
or actually, i don't like either. in systemd we have this rule that functions should either log about all their errors, or about none (where logging means: log above LOG_DEBUG log level, because LOG_DEBUG logging is always ok). 

But these two new errors here you do not log about, though you do log about all other errors. 

hence, maybe change the gather_pid_mount_tree_fd() prototype to this:

```c
static int gather_pid_mount_tree_fd(const Context *context, int *ret_fd) {
```

And now, change these two EOPNOTSUPP/EHOSTDOWN cases to return zero (i.e. success), but then return with `*ret_fd = -EBADF;`. Thus, things are reasonably clean: we treat all errors as errors, and whether we return an fd or not on success is up to us.